### ISSUE TITLE
isEmpty tries to run twice

### ISSUE THREAD

**Issue by lfrancke** [solven-eu/cleanthat#485]
Sorry, this is only from my phone again. I can try and add more information later.

Using cleanthat 2.13 from spotlessaven plugin 2.35

It finds instances of "foo.size() == 0” and wants to replace them with isEmpty.

But it finds every occurrence twice. It then replaces one and the second one then gives a warning "We failed turning X into Y"

The end result is correct.

**Comment by blacelle** [comment]
Thanks @lfrancke , I've been able to reproduce. A plain:

		if (new ArrayList<>().size() == 0) {
			return true;
		}

will indeed log:

```
[INFO] Turning `new ArrayList<>().size() == 0` into `new ArrayList<>().isEmpty()`
[INFO] Turning `new ArrayList<>().size() == 0` into `new ArrayList<>().isEmpty()`
[WARNING] We failed turning `new ArrayList<>().size() == 0` into `new ArrayList<>().isEmpty()`
```

**Comment by blacelle** [comment]
The issue seems not restricted to `UseCollectionIsEmpty` mutator, but it is related to a sanityCheck (which check it failed mutating twice the same node). The idem-potency sanityCheck succeeds, but it trigger another sanityCheck: on the secondTry, the node has no parent anymore (as it has been replaced), then the replace operation fails.

-> These logs are garbage.

@lfrancke This makes me feel you faced a lower number of replacements in your run. Knowing a bit your context, you may be interested in more advanced refactoring rules, as available in :

- SafeAndConsensual (the default one, probably activated in your case)
- SafeButNotConsensual (you may wish to disable some mutators amongst then)
- SafeButControversial (this would apply more advanced refactoring (e.g. converting for-each loops into Stream API))

(https://github.com/solven-eu/cleanthat/blob/master/MUTATORS.generated.MD#safebutnotconsensualmutators)

**Comment by lfrancke** [comment]
Thanks! I'm just using the defaults from spotless. It says "*" there, I'm not sure if that really means all of them or if that will be `SafeAndConsensual`?

**Comment by blacelle** [comment]
This is old (and confusing) syntax I'll update asap.

Spotless default is `SafeAndConsensual`. I need to find a way to easily activate more rules as some people struggle catching what can be achieved with cleanthat.

You may also be interested activating `includeDraft`, as current set of prd-rules is very conservative (I'll soon dive into a multi-projects validation pass).

**Comment by lfrancke** [comment]
I'm fine with _very conservative_ :)
Thank you for the update.

**Comment by blacelle** [comment]
Should be fixed in 2.17
